services/graph/pkg/service/v0/base.go

```diff
@@ -482,6 +482,10 @@ func (g BaseGraphService) cs3UserShareToPermission(ctx context.Context, share *c
 		perm.SetRoles([]string{role.GetId()})
 	} else {
 		actions := unifiedrole.CS3ResourcePermissionsToLibregraphActions(share.GetPermissions().GetPermissions())
+		// neither a role nor actions are set, that is a full denial
+		if len(actions) == 0 {
+			actions = []string{"access denied"}
```

Comment:
@rhafer Not the "nicest" solution, I admit.

We have the contract with the clients that we 
- either set a value on `roles`
- or set a value on `@libre.graph.permissions.actions`

That means, if we disable the role "RoleDenied" and have existing shares, we need to return at least a hint to the user.

Fine for me, if you want to solve it differently.

Reply:
After discussion with @butonic and @kobergj we decided to return `none` in that case.